### PR TITLE
Add support for appearance prop to the Alert component

### PR DESCRIPTION
**Pull request by rohitpagariya**
This PR adds support to render the Alert component in primary & inverted styles via the `appearance` prop

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                       | Baseline (minified/GZIP) |                          PR |       Change |
| :-------------------------------------------------------------------------------------- | -----------------------: | --------------------------: | -----------: |
| <samp>react-alert</samp> <br /> <abbr title='bundle-size/Alert.fixture.js'>Alert</abbr> |         `0 B`<br />`0 B` | `80.66 kB`<br />`20.482 kB` | 🆕 New entry |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `189.067 kB`<br />`52.144 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `31.883 kB`<br />`10.516 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/e81a3de6b292b226e09ad8526ae5856131ff0736'>e81a3de6b292b226e09ad8526ae5856131ff0736</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23780/perf-test-react-components/Avatar-mount_base.html">1386</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23780/perf-test-react-components/Avatar-mount.html">1368</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23780/perf-test-react-components/Button-mount_base.html">1022</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23780/perf-test-react-components/Button-mount.html">1022</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23780/perf-test-react-components/FluentProvider-mount_base.html">1771</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23780/perf-test-react-components/FluentProvider-mount.html">1747</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23780/perf-test-react-components/FluentProviderWithTheme-mount_base.html">727</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23780/perf-test-react-components/FluentProviderWithTheme-mount.html">698</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23780/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">657</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23780/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">662</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23780/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">707</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23780/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">750</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23780/perf-test-react-components/MakeStyles-mount_base.html">2116</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23780/perf-test-react-components/MakeStyles-mount.html">2150</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by ling1726** [comment]
@rohitpagariya  could you please resolve the conflicts ?

File: packages/react-components/react-avatar/MIGRATION-AvatarGroup.md [hunk 1]
```diff
@@ -11,8 +11,9 @@ AvatarGroup and Facepile have similar APIs, but the main difference is how they
 Avatars must not be used inside an AvatarGroup, instead AvatarGroupItems are used since it has extra functionality used only in AvatarGroups. This includes adding a label when the AvatarGroupItem is rendered in the overflow Popover and specific styling for each layout.
 
 AvatarGroup has extra functionality that Facepile does not have, this includes:
- - When the AvatarGroupItems are overflowing, a Popover is rendered that contains all the overflowed AvatarGroupItems. The Popover is triggered when the overflow button is clicked.
- - AvatarGroup supports three layouts: `spread` (default), `stack`, and `pie`.
+
```
Comment (spmonahan) [inline packages/react-components/react-avatar/MIGRATION-AvatarGroup.md:14]: Is this change intentional? I wouldn't expect any changes to Avatar as part of this PR.

File: packages/react-components/react-avatar/README-AvatarGroup.md [hunk 2]
```diff
@@ -1,39 +1,39 @@
...
-  <AvatarGroupItem name="Cameron Evans" />
-  <AvatarGroupItem name="Wanda Howard" />
-  <AvatarGroupItem name="Mona Kane" />
-  <AvatarGroupItem name="Allan Munger" />
-  <AvatarGroupItem name="Daisy Phillips" />
-  <AvatarGroupItem name="Robert Tolbert" />
-  <AvatarGroupItem name="Kevin Sturgis" />
-</AvatarGroup>
-```
+# @fluentui/react-avatar
```
Comment (spmonahan) [inline packages/react-components/react-avatar/README-AvatarGroup.md:1]: Do you know what changed in this file? Whitespace maybe?

Comment (sopranopillow) [reply packages/react-components/react-avatar/README-AvatarGroup.md:1]: @rohitpagariya do you mind restoring these AvatarGroup files?

Comment (rohitpagariya) [reply packages/react-components/react-avatar/README-AvatarGroup.md:1]: No it's not. It's a whitespace change & I can't seem to undo it unfortunately.

File: packages/react-components/react-avatar/MIGRATION-AvatarGroup.md (hunk 1 above)
Comment (rohitpagariya) [reply packages/react-components/react-avatar/MIGRATION-AvatarGroup.md:14]: It's a whitespace change & I can't seem to undo it unfortunately.

File: packages/react-components/react-avatar/README-AvatarGroup.md (hunk 2 above)
Comment (sopranopillow) [reply packages/react-components/react-avatar/README-AvatarGroup.md:1]: You can checkout from master, then commit with the flag --no-verify and that should work

Comment (rohitpagariya) [reply packages/react-components/react-avatar/README-AvatarGroup.md:1]: Nope, didn't work. Deleting & restoring didn't work either. I can even tell what's the diff tbh.

Comment (rohitpagariya) [reply packages/react-components/react-avatar/README-AvatarGroup.md:1]: Mind giving a try?

File: packages/react-components/react-alert/src/components/Alert/useAlertStyles.ts [hunk 3]
```diff
@@ -60,9 +65,15 @@ const useIntentIconStyles = makeStyles({
...
 export const useAlertStyles_unstable = (state: AlertState): AlertState => {
   const styles = useStyles();
+  const appearance = state.appearance || 'primary';
   const intentIconStyles = useIntentIconStyles();
 
-  state.root.className = mergeClasses(alertClassNames.root, styles.root, state.root.className);
+  state.root.className = mergeClasses(
+    alertClassNames.root,
+    styles.root,
+    appearance !== 'primary' && styles.invertedRoot,
```
Comment (layershifter) [inline packages/react-components/react-alert/src/components/Alert/useAlertStyles.ts:74]:
```suggestion
    appearance !== 'primary' && styles.inverted,
```

IMO we don't use "root" suffix for styles defined for "root" slot

File: packages/react-components/react-alert/src/stories/Alert/AlertAppearance.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+const renderAlerts = (appearance: 'primary' | 'inverted') => (
+  <div style={{ flexGrow: 1 }}>
+    <Title3>{appearance}</Title3>
+    <div style={{ display: 'flex', flexDirection: 'column', gap: '10px' }}>
+      <Alert
+        appearance={appearance}
+        intent="success"
+        action={{
+          children: 'Undo',
+        }}
```
Comment (layershifter) [inline packages/react-components/react-alert/src/stories/Alert/AlertAppearance.stories.tsx:14]:
```suggestion
        action='Undo'
```

nit: You can use shorthand form for this slot as other props are not used

**Review by layershifter — APPROVED** [review]

**Review by ling1726 — APPROVED** [review]

File: packages/react-components/react-avatar/README-AvatarGroup.md (hunk 2 above)
Comment (sopranopillow) [reply packages/react-components/react-avatar/README-AvatarGroup.md]: done! if this problem comes up again you can do this from root:
`git checkout upstream/master packages/react-components/react-avatar/* && git commit --no-verify -m "{your message}"` and that should restore it to the original file.

File: packages/react-components/react-alert/src/components/Alert/Alert.test.tsx [hunk 5]
```diff
@@ -47,7 +47,11 @@ describe('Alert', () => {
   });
 
   it('renders a button', () => {
-    render(<Alert action={{ children: 'Undo' }}>Test</Alert>);
+    render(
+      <Alert action={{ children: 'Undo' }} appearance="inverted">
```
Comment (khmakoto) [inline packages/react-components/react-alert/src/components/Alert/Alert.test.tsx:51]:
```suggestion
      <Alert action="Undo" appearance="inverted">
```

nit: Similar to @layershifter's comment below, you can use shorthand form for this slot as other props are not used

File: packages/react-components/react-alert/src/components/Alert/useAlert.tsx [hunk 6]
```diff
@@ -52,21 +52,22 @@ export const useAlert_unstable = (props: AlertProps, ref: React.Ref<HTMLElement>
   }
 
   return {
+    action: resolveShorthand(props.action, { defaultProps: { appearance: 'transparent' } }),
```
Comment (khmakoto) [inline packages/react-components/react-alert/src/components/Alert/useAlert.tsx:55]: _nit:_ Can you move this next to the other resolved shorthands?

File: packages/react-components/react-alert/src/components/Alert/Alert.types.ts [hunk 7]
```diff
@@ -31,9 +31,14 @@ export type AlertProps = ComponentProps<AlertSlots> & {
...
+   * @default 'primary'
+   */
+  appearance?: 'primary' | 'inverted';
 };
 
 /**
  * State used in rendering Alert
  */
-export type AlertState = ComponentState<AlertSlots> & Pick<AlertProps, 'intent'>;
+export type AlertState = ComponentState<AlertSlots> & Pick<AlertProps, 'appearance' | 'intent'>;
```
Comment (khmakoto) [inline packages/react-components/react-alert/src/components/Alert/Alert.types.ts:44]: Should we make appearance required in state as we always default to primary if it's not present?

```suggestion
export type AlertState = ComponentState<AlertSlots> & Pick<AlertProps, 'intent'> & Required<Pick<AlertProps, 'appearance'>>;
```

File: packages/react-components/react-alert/src/components/Alert/useAlertStyles.ts [hunk 8]
```diff
@@ -60,9 +65,15 @@ const useIntentIconStyles = makeStyles({
  */
 export const useAlertStyles_unstable = (state: AlertState): AlertState => {
   const styles = useStyles();
+  const appearance = state.appearance || 'primary';
```
Comment (khmakoto) [inline packages/react-components/react-alert/src/components/Alert/useAlertStyles.ts:68]: If we make appearance required in state you wouldn't need to do this and you could just use `state.appearance` directly below.

**Review by khmakoto — APPROVED** [review]

**Review by sopranopillow — APPROVED** [review]
